### PR TITLE
`Development`: Fix and re-enable exam mode e2e tests

### PR DESCRIPTION
**Pull request by SimonEntholzer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] This change only affects tests.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

The exam mode e2e tests are broken, as a new UI was introduced.

### Description

This PR fixes the outdated exam mode e2e tests.

### Steps for Testing

Prerequisites:
- make sure that all playwright tests pass (see below, multi/single node.., run locally)

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests (check out the test logs below)
- [ ] Test 1
- [ ] Test 2

## Summary by CodeRabbit

- **New Features**
  - Improved navigation and interaction within exams, including refined URL handling and timing validations.
  - Enhanced exam participation flow using the navigation sidebar and exercise titles.
  - Added methods for editing and deleting test exam runs.

- **Bug Fixes**
  - Fixed issues with exercise opening and saving logic during exams.
  - Corrected URL patterns and navigation steps for exam-related pages.

- **Refactor**
  - Replaced index-based exercise navigation with title-based navigation.
  - Adjusted method names and parameters for better clarity and functionality.

- **Chores**
  - Removed outdated `testIgnore` configuration for exam tests in Playwright configuration.

### CODE REVIEW

**Review by egekurt123 — DISMISSED** [review]
Code LGTM

File: src/test/playwright/e2e/exam/ExamParticipation.spec.ts [hunk 1]
```diff
@@ -109,12 +107,16 @@ test.describe('Exam participation', () => {
...
+
                 // Skip programming exercise this time to save execution time
                 // (we also need to use the navigation bar here, since programming  exercises do not have a "Save and continue" button)
+
                 if (exercise.type != ExerciseType.PROGRAMMING) {
-                    await examNavigation.openExerciseOverview();
-                    await examParticipation.selectExerciseOnOverview(j + 1);
-                    await examParticipation.makeSubmission(exercise.id!, exercise.type!, exercise.additionalData);
+                    // await examNavigation.openExerciseFromOverviewPageByTitle(exercise.title!)
+                    // await examParticipation.makeSubmission(exercise.id!, exercise.type!, exercise.additionalData);
```
Comment (JohannesStoehr) [inline src/test/playwright/e2e/exam/ExamParticipation.spec.ts:119]:
```suggestion
```

File: src/test/playwright/e2e/exam/ExamTestRun.spec.ts [hunk 2]
```diff
@@ -109,16 +109,16 @@ test.describe('Exam test run', () => {
...
             await examParticipation.handInEarly();
             for (let j = 0; j < exerciseArray.length; j++) {
                 const exercise = exerciseArray[j];
                 await examParticipation.verifyExerciseTitleOnFinalPage(exercise.id!, exercise.exerciseGroup!.title!);
-                if (exercise.type === ExerciseType.TEXT) {
-                    await examParticipation.verifyTextExerciseOnFinalPage(exercise.id!, exercise.additionalData!.textFixture!);
-                }
+                // if (exercise.type === ExerciseType.TEXT) {
+                //     await examParticipation.verifyTextExerciseOnFinalPage(exercise.id!, exercise.additionalData!.textFixture!);
+                // }
```
Comment (JohannesStoehr) [inline src/test/playwright/e2e/exam/ExamTestRun.spec.ts:121]:
```suggestion
```

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Additionally a lot of exam tests seam to fail: https://bamboo.ase.in.tum.de/browse/ARTEMIS-AEPTMA585-2/test
Is this related to this PR?

**Comment by SimonEntholzer** [comment]
> Additionally a lot of exam tests seam to fail: https://bamboo.ase.in.tum.de/browse/ARTEMIS-AEPTMA585-2/test Is this related to this PR?



> Additionally a lot of exam tests seam to fail: https://bamboo.ase.in.tum.de/browse/ARTEMIS-AEPTMA585-2/test Is this related to this PR?

Yea some tests were not fixed yet at that point

File: src/test/playwright/e2e/exam/ExamParticipation.spec.ts (hunk 1 above)
Comment (SimonEntholzer) [reply src/test/playwright/e2e/exam/ExamParticipation.spec.ts]: removed the code :+1:

File: src/test/playwright/e2e/exam/ExamDateVerification.spec.ts [hunk 3]
```diff
@@ -112,14 +106,15 @@ test.describe('Exam date verification', () => {
             await examAPIRequests.generateMissingIndividualExams(exam);
             await examAPIRequests.prepareExerciseStartForExam(exam);
             await login(studentOne);
-            await page.goto(`/courses/${course.id}/exams`);
-            await courseOverview.openExam(exam.id!);
+            await page.goto(`/courses/${course.id}/exams/${exam.id}`);
+            await page.waitForURL(`**/exams/${exam.id}`);
```
[the remark this answers is not in the document]
Comment (SimonEntholzer) [reply src/test/playwright/e2e/exam/ExamDateVerification.spec.ts:110]: Pleas do not provide these "safer alternative" tips for this PR anymore

**Pull request merged**